src/frontend/js/contexts/OrderContext/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useMemo, useState } from 'react';
+import { Order } from '../../types/Joanie';
+import { Maybe } from '../../types/utils';
+
+export interface OrderContextInterface {
+  currentOrder: Maybe<Order>;
+  setCurrentOrder: (order: Maybe<Order>) => void;
+}
+
+export const OrderContext = createContext<Maybe<OrderContextInterface>>(undefined);
+
+export const useCurrentOrderContext = () => {
+  const orderContext = useContext(OrderContext);
+
+  if (orderContext) {
+    return orderContext;
+  }
+
+  throw new Error(`orderContext must be used within a OrderContext`);
+};
+
+export interface OrderProviderProps {
+  order: Maybe<Order>;
+}
+
+export const OrderProvider = ({ order, children }: PropsWithChildren<OrderProviderProps>) => {
+  const [currentOrder, setCurrentOrder] = useState(order);
+  const value = useMemo(
+    () => ({
+      currentOrder,
+      setCurrentOrder,
+    }),
+    [currentOrder],
+  );
+
+  useEffect(() => {
+    setCurrentOrder(order);
+  }, [order]);
+
+  return <OrderContext.Provider value={value}>{children}</OrderContext.Provider>;
+};
```

Comment:
To respect other context convention I would revert order of method declaration. Furthermore other context export by default the OrderProvider.
```
const OrderContext = ...
const OrderProvider = ...
export const useOrderContext = ...

export default OrderProvider
```